packages/SwingSet/test/test-liveslots.js

```diff
@@ -1394,7 +1597,7 @@ test('promise cycle', async t => {
   t.deepEqual(log, []);
 });
 
-test('unserializable promise resolution', async t => {
+test.serial('unserializable promise resolution', async t => {
```

Comment:
do these need to be serial?

Reply:
I'm not 100% sure _why_, but yes.  If you run the tests together these tests now fail, but if you run them individually the succeed.

Reply:
Ah, I found it, the new test uses `setSyscallCapdataLimits()` to reduce the size limit to an artificially low value, and those limits are kept in a module-level `let` binding so they affect all callers in the same process (and it doesn't set the size back again after the test is done). That limit is low enough to cause the other tests to fail from a "capdata too big" error instead of the other errors they were trying to exercise.

As supporting evidence, running `yarn test -s test/test-liveslots.js` also fails (the `-s` effectively adds `.serial` to every `test()` in the file). I'm guessing that if you only mark a few tests as `.serial`, they get run first, so these two tests were getting to run before the size limit was reduced.

We need those size limits to be per-liveslots-instance instead of being at module level. Just move the `let`s and the checker function inside the definition of `makeLiveSlots()`.

Reply:
Fixed the scoping.